adapters/deepintent/deepintent.go

```diff
@@ -0,0 +1,142 @@
+package deepintent
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// DeepintentAdapter struct
+type DeepintentAdapter struct {
+	URI string
+}
+
+// NewDeepintentBidder Initializes the Bidder
+func NewDeepintentBidder(endpoint string) *DeepintentAdapter {
+	return &DeepintentAdapter{
+		URI: endpoint,
+	}
+}
+
+func (d *DeepintentAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+
+	var errs []error
+
+	reqJSON, err := json.Marshal(request)
+
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     d.URI,
+		Body:    reqJSON,
+		Headers: headers,
+	}, errs
+}
+
+type deepintentParams struct {
+	TagID string `json:"TagID"`
+}
+
+// MakeBids makes the bids
+func (d *DeepintentAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusNotFound {
+		return nil, []error{&errortypes.BadServerResponse{
```

Comment:
Please add test coverage for this condition.

Reply:
Added test coverage